The gnu-libiconv test is solely for platforms where the inbuf parameter to
iconv(3) is const char **; we have GNU libiconv but use the POSIX char **.

--- origsrc/qt-everywhere-opensource-src-4.8.5/config.tests/unix/iconv/iconv.pro	2013-06-07 00:16:41.000000000 -0500
+++ src/qt-everywhere-opensource-src-4.8.5/config.tests/unix/iconv/iconv.pro	2013-08-20 15:03:32.020470300 -0500
@@ -1,3 +1,3 @@
 SOURCES = iconv.cpp
 CONFIG -= qt dylib app_bundle
-mac|win32-g++*|qnx:LIBS += -liconv
+mac|cygwin-g++*|win32-g++*|qnx:LIBS += -liconv
